pyccel/codegen/printing/ccode.py

```diff
@@ -323,76 +316,53 @@ def arrayFill(self, expr):
         ------
             String
                 Return a str that contains a call to the C function array_fill,
-                if the ndarray is a stack_array the str will contain the initialization
         """
         rhs = expr.rhs
         lhs = expr.lhs
         code_init = ''
         declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
 
-        if lhs.is_stack_array:
-            symbol_map = {}
-            used_names_tmp = self._parser.used_names.copy()
-            sympy_shapes = [pyccel_to_sympy(s, symbol_map, used_names_tmp) for s in lhs.alloc_shape]
-
-            length = functools.reduce(operator.mul, sympy_shapes)
-            printable_length = functools.reduce(PyccelMul, lhs.alloc_shape)
-
-            length_code = self._print(printable_length)
-
-            if length.is_constant():
-                buffer_array = "({declare_dtype}[{length}]){{}}".format(
-                                        declare_dtype = declare_dtype,
-                                        length=length_code)
-            else:
-                dummy_array_name, _ = create_incremented_string(self._parser.used_names,
-                                                                prefix = lhs.name+'_data')
-                code_init += "{dtype} {name}[{length}];\n".format(
-                        dtype  = declare_dtype,
-                        name   = dummy_array_name,
-                        length = length_code)
-                buffer_array = dummy_array_name
-
-            code_init += self._init_stack_array(expr, buffer_array)
-
         if rhs.fill_value is not None:
             if isinstance(rhs.fill_value, Literal):
                 code_init += 'array_fill(({0}){1}, {2});\n'.format(declare_dtype, self._print(rhs.fill_value), self._print(lhs))
             else:
                 code_init += 'array_fill({0}, {1});\n'.format(self._print(rhs.fill_value), self._print(lhs))
         return code_init
 
-    def _init_stack_array(self, expr, buffer_array):
+    def _init_stack_array(self, expr):
         """ return a string which handles the assignment of a stack ndarray
 
         Parameters
         ----------
             expr : PyccelAstNode
                 The Assign Node used to get the lhs and rhs
-            buffer_array : String
-                The data buffer
         Returns
         -------
             Returns a string that contains the initialization of a stack_array
         """
```

Comment:
There are now two different return values, so the docstring should be updated accordingly

Reply:
Fixed 16f3bab2